### PR TITLE
Bedrock embedding query for cohere was not in the correct format

### PR DESCRIPTION
**Pull request by deuscapturus**
# Description

Fix: Bedrock get embedding was not formatting the request body properly for cohere models.  Instead of passing the valid json text format to the boto client, it was passing a json.parse() or whatever string was passed to `get_text_embedding`.

This also adds a unimplemented feature to get embeddings for queries as well and fixes type signatures for internal methods.

Example:
```
embedding = BedrockEmbedding(
    client=bedrock_client,
    model_name=Models.COHERE_EMBED_ENGLISH_V3,
)

embedding.get_text_embedding("some text")
```

Results in error:
```
botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the InvokeModel operation: Malformed input request, please reformat your input and try again.
```

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by hatianzhang** [comment]
awesome! will take a look

**Comment by hatianzhang** [comment]
@deuscapturus pls help fix the unit tests and lint errors

File: tests/embeddings/test_bedrock.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import boto3
+from botocore.response import StreamingBody
+from botocore.stub import Stubber
+from llama_index.embeddings.bedrock import BedrockEmbedding, Models
+
+
+class TestBedrockEmbedding(TestCase):
+    bedrock_client = boto3.client("bedrock-runtime", region_name="us-east-1")
+    bedrock_stubber = Stubber(bedrock_client)
```
[the remark this answers is not in the document]
Comment (deuscapturus) [reply tests/embeddings/test_bedrock.py:13]: Looking at llms/test_bedrock.py it seems that tests are just skipped if boto is not available.  I recommend using the boto Stubber.  It does some parameter validation and offers better test coverage against AWS.

I can add boto Stubber to `llms/test_bedrock.py` for better coverage as well.

[the remark this answers is not in the document]
Comment (deuscapturus) [reply tests/embeddings/test_bedrock.py:13]: @hatianzhang While implementing the boto Stubber on `tests/llms/test_bedrock.py` I did find what looks like other issues with the Mocks used in those tests.

Let me know if you are firm on not including boto as a dev dependency.  If you are open to it I will fix llms/test_bedrock.py as well in this PR

[the remark this answers is not in the document]
Comment (hatianzhang) [reply tests/embeddings/test_bedrock.py:13]: @deuscapturus thanks 

I am okay to include to boto as a pkg. If possible, pls put a fix for `llms/test_bedrock.py` as well which will be awesome

**Review by hatianzhang — APPROVED** [review]
LGTM! This is great @deuscapturus !

**Pull request merged**